### ISSUE TITLE
Tableau extensions running on localhost or invalid certificates no longer allowed in Tableau 2020.4.2?

### ISSUE THREAD

**Issue by bggoxp** [tableau/extensions-api#370]
Hi everyone,

In Tableau Desktop version 2020.4.2 Tableau is not allowing extensions to run on localhost or with an invalid certificate; This was not an issue in Tableau version 2020.4.1 and I could not find anything related to this on the release notes page. Could you please provide any details regarding this issue? 
With this change, the extension is not allowed to load at all. Is this intended?
I can see this error in the console and that's it:
![image](https://user-images.githubusercontent.com/47524467/110136353-8055c280-7dc7-11eb-9e1c-083505dcd584.png)

Thanks

**Comment by johnDance** [comment]
We think this one is the result of recent browser changes. Can you navigate to the url in chrome and accept the self-signed cert?That should then allow it elsewhere.
Let us know if that doesn't work.
John

**Comment by bggoxp** [comment]
Hi John,
Thanks for the quick reply.
What you suggested works when using on a browser, like Tableau online or Tableau server but this issue is happening in Tableau Desktop where there is no way to do that.

With previous versions, we could simply install the certificate locally and it would work but now that won't do it.

**Comment by seanmakesgames** [comment]
Hey @bggoxp --

I know it sounds weird, but chrome stores its security permissions in a way that's shared across instances of the browser. Tableau embeds chrome in desktop, so when you go into the chrome browser in a different window and navigate and accept the certificate, it sometimes allows it to work inside Tableau desktop.

After you try that and if it doesn't work, can you share with us which network request is getting gated by this? You should be able to see this in the network tab of the chrome devtools.

Once we know that, we can know better which cert is failing.

**Comment by bggoxp** [comment]
Hi @seanmakesgames,

Yeah, I've tried that already and it still does not fix it. 

The request is just the very first one to the server, as I posted in the first post, that error is thrown when contacting the server. The cert failing is the self-signed one that I have on my local server.

Thanks.

**Comment by bggoxp** [comment]
I did some further research and noticed that in version 2020.2 it happened after upgrading from 2020.2.6 to 2020.2.7.

**Comment by seanmakesgames** [comment]
Can you try this out with your certificate / verify that this is the case already?
https://stackoverflow.com/questions/43665243/invalid-self-signed-ssl-cert-subject-alternative-name-missing

This is the behavior I'm seeing on my machine when I repro it.

**Comment by seanmakesgames** [comment]
I have verified on my machine that providing the SAN in the extended certificate parameters has fixed the issue in desktop.
Please try this with your self-signed certificate and let us know.

**Comment by bggoxp** [comment]
Hi, thank you very much. That does solve the issue!

**Comment by seanmakesgames** [comment]
> Hi, thank you very much. That does solve the issue!

Thanks so much for letting us know!
